frontend/components/header.js

```diff
@@ -16,18 +16,39 @@ class Header extends HTMLElement {
   connectedCallback() {
     const rawUsername = localStorage.getItem('signedIn') === "true" ? localStorage.getItem('username') : "there";
     const username = escapeHTML(rawUsername);
-
+    const profilePicture = username === 'there' ? `assets/images/default-avatar.jpeg` : `https://github.com/${username}.png`;
+    const links = localStorage.getItem('signedIn') === "true" 
+                ? ` <a href="leaderboard.html">leaderboard</a>
+                    <a href="attempts.html">My attempts</a>
+                    <a href="logout.html">logout</a>`
+                : `<a href="${Links.serverBaseURL}/auth/github">login</a>
+                   `
     this.innerHTML = `
-        <header>
-            <img class="logo" src=${Icons.logo} alt="logo" />
-  
-            <aside>
-                <img class="moon" id="theme-toggle-btn" src=${Icons.moon} alt="theme-toggler" />
-                <h6>Hello, ${username}</h6>
-            </aside>
-        </header>
+            <header>
+                <img class="logo" src=${Icons.logo} alt="logo" />
+                <div class="dropdown">
+                    <img src=${profilePicture} alt="Avatar" class="avatar">
+                    <div class="dropdown-content">
+                        <h6 class="dropdown-name">Hello, ${username}</h6>
+                        <div class="dropdown-links">
+                            <a href="#" id="theme-toggle-btn"> 
+                                    <img class="moon" src=${Icons.moon} alt="theme-toggler" />
+                                    <span>dark mode</span>
```

Comment:
Span is not really a semantic element, we might be questioned why we used it. i would suggest using a <p> tag

Reply:
I think span is justified here, basically it's there for flex to work properly